components/kvengine/src/table/sstable/builder.rs

```diff
@@ -417,8 +417,11 @@ impl BlockBuilder {
         for i in 0..num_entries {
             self.block.build_entry(buf, i, common_prefix_len);
         }
-        if self.compression_tp != NO_COMPRESSION {
-            self.compress();
+        match self.compression_tp {
+            NO_COMPRESSION => (),
+            LZ4_COMPRESSION => self.compress_lz4(),
+            ZSTD_COMPRESSION => self.compress_zstd(),
+            _ => panic!("unexpected compression type {}", self.compression_tp),
```

Comment:
It's better to use enum to represent compression type. I can refactor it later.

Reply:
Yes, we can do it in another PR.